contracts/governance/Staking/Staking.sol

```diff
@@ -490,6 +557,28 @@ contract Staking is IStaking, WeightedStaking, ApprovalReceiver {
 		_moveDelegates(currentDelegate, delegatee, delegatorBalance, lockedTS);
 	}
 
+	// @dev delegates tokens for lock date 2 weeks later than given lock date
+	//		if message sender is a contract
+	function _delegateNext(
+		address delegator,
+		address delegatee,
+		uint256 lockedTS
+	) internal {
+		if (msg.sender.isContract()) {
```

Comment:
anyone can deploy a contract to get access to that logic. perform a more specific check

Reply:
added function _isVestingContract